yb-voyager/cmd/endMigrationCommand.go

```diff
@@ -0,0 +1,418 @@
+}
+
+func backupSchemaFilesFn() {
+	if !bool(backupSchemaFiles) || !utils.FileOrFolderExists(filepath.Join(exportDir, "schema")) {
+		return
+	}
+
+	utils.PrintAndLog("backing up schema files")
+	cmd := exec.Command("mv", filepath.Join(exportDir, "schema"), backupDir)
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		utils.ErrExit("end migration: moving schema files: %s: %v", string(output), err)
+	}
+}
+
+func backupDataFilesFn() {
+	if !backupDataFiles {
+		return
+	}
+
+	utils.PrintAndLog("backing up snapshot sql data files")
+	err := os.MkdirAll(filepath.Join(backupDir, "data"), 0755)
+	if err != nil {
+		utils.ErrExit("end migration: creating data directory for backup: %v", err)
+	}
+
+	files, err := os.ReadDir(filepath.Join(exportDir, "data"))
+	if err != nil {
+		utils.ErrExit("end migration: reading data directory: %v", err)
+	}
+	for _, file := range files {
+		if file.IsDir() || !strings.HasSuffix(file.Name(), ".sql") {
+			continue
+		}
+
+		err = os.Rename(filepath.Join(exportDir, "data", file.Name()), filepath.Join(backupDir, "data", file.Name()))
+		if err != nil {
+			utils.ErrExit("end migration: moving data files: %v", err)
+		}
+	}
+}
+
+func saveMigrationReportsFn() {
+	if !saveMigrationReports {
+		return
+	}
+
+	// TODO: what if there is no report.txt generated from analyze-schema step
+	utils.PrintAndLog("saving schema analysis report")
+	files, err := os.ReadDir(filepath.Join(exportDir, "reports"))
+	if err != nil {
+		utils.ErrExit("end migration: reading reports directory: %v", err)
+	}
+	for _, file := range files {
+		if file.IsDir() || !strings.HasPrefix(file.Name(), "report.") {
+			continue
+		}
+
+		err = os.Rename(filepath.Join(exportDir, "reports", file.Name()), filepath.Join(backupDir, "reports", file.Name()))
+		if err != nil {
+			utils.ErrExit("end migration: moving migration reports: %v", err)
+		}
+	}
+
+	utils.PrintAndLog("saving data export reports...")
+	file, err := os.Create(filepath.Join(backupDir, "reports", "export_data_report.txt"))
+	if err != nil {
+		utils.ErrExit("end migration: creating export data status report: %v", err)
+	}
+	exportDataStatusCmd := exec.Command("bash", "-c", fmt.Sprintf("yb-voyager export data status -e %s", exportDir))
+	exportDataStatusCmd.Stdout = file
+	var outbuf bytes.Buffer
+	exportDataStatusCmd.Stderr = &outbuf
+	err = exportDataStatusCmd.Run()
+	if err != nil {
+		log.Errorf("end migration: running export data status command: %s: %v", outbuf.String(), err)
+		utils.ErrExit("end migration: running export data status command: %v", err)
+	}
+	file.Close()
+
+	utils.PrintAndLog("saving data import reports...")
+	file, err = os.Create(filepath.Join(backupDir, "reports", "import_data_report.txt"))
+	if err != nil {
+		utils.ErrExit("end migration: creating import data status report: %v", err)
+	}
+	importDataStatusCmd := exec.Command("bash", "-c", fmt.Sprintf("yb-voyager import data status -e %s", exportDir))
+	importDataStatusCmd.Stdout = file
```

Comment:
Because we're invoking `bash`, we can also use shell's stdout redirection here:
```
strCmd := fmt.Sprintf("yb-voyager import data status -e %s > %q", exportDir, importDataReportFilePath)
cmd := exec.Command("bash", "-c", strCmd)
errBuf := bytes.Buffer{}
cmd.Stderr = errBuf
...
```